Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/ApiGatewayRequestTranslatorFactory.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace Amazon.Lambda.TestTool
+{
+
+    /// <summary>
+    /// Factory class for creating API Gateway request translators based on the specified API Gateway mode.
+    /// </summary>
+    public class ApiGatewayRequestTranslatorFactory : IApiGatewayRequestTranslatorFactory
```

Comment:
I think this is a bit overkill. The catch all api will decide which function to call based on the mode if we use extension methods.